compiler/src/iree/compiler/Codegen/SPIRV/Passes.cpp

```diff
@@ -564,7 +564,7 @@ void addSPIRVSubgroupReducePassPipeline(OpPassManager &pm) {
   nestedModulePM.addNestedPass<func::FuncOp>(createCSEPass());
 
   // Bufferize and distribute.
-  addSPIRVBufferizePasses(nestedModulePM, gpuAllocateFunctionMemoryFn);
+  addSPIRVBufferizePasses(nestedModulePM, gpuAllocateWorkgroupMemoryFn);
```

Comment:
Can you add the comment in the PR description here as well? Explaining why it needs to be shared memory.

Reply:
Good point! Done.